contrib/devtools/github-merge.py

```diff
@@ -175,14 +200,29 @@ def main():
             print("ERROR: Creating merge failed (already merged?).",file=stderr)
             exit(4)
 
+        # Put tree SHA512 into the message
+        try:
+            first_sha512 = tree_sha512sum()
+            message += '\n\nTree-SHA512: ' + first_sha512
+        except subprocess.CalledProcessError as e:
+            printf("ERROR: Unable to compute tree hash")
+            exit(4)
+        try:
+            subprocess.check_call([GIT,'commit','--amend','-m',message.encode('utf-8')])
+        except subprocess.CalledProcessError as e:
+            printf("ERROR: Cannot update message.",file=stderr)
+            exit(4)
+        second_sha512 = tree_sha512sum()
```

Comment:
Why is the above call to tree_sha512sum() guarded by a try except clause and this one is not?

Anyway, I think the right place to put the second check would be `verify-commits.sh`.